CONTRIBUTING.md

```diff
@@ -238,4 +238,49 @@ regression    =   1.44 sec*proc (1 test)
 Total Test time (real) =   1.47 sec
 ```
 
+#### Including Tests in Code Coverage Report
+
+Tests are not automatically included in the coverage report. To include a unit or integration test in the coverage
+report the tag [coverage] should be added to the catch2 `TEST_CASE` macro. E.g.
+
+```
+TEST_CASE("test description", "[unit][coverage]"){
+...
+}
+```
+
+To add a regression test to the coverage report an attribute of the parameters argument can be 
+altered. E.g.
+
+```
+class TestCase(utils.test_case.TestCaseAbs):
+    def Prepare(self,parameters):
+        parameters.coverage_status = "only-coverage"
+        return parameters
+```
+
+The coverage status argument can take 3 different settings:
+
+1. **only-regression** - will not run if with coverage, is the default
+2. **both** - will run with or without coverage
+3. **only-coverage** - will only run if coverage is specified
```

Comment:
I don't love that we have to add coverage in two separate spots. I don't see a way around this though.

Reply:
Just to be clear, you don't need to add it in two different spots for any given test. It's just that if you use a regression test you are adding it in python and in a unit test you are adding it in the cpp test files.